src/main/java/cl/transbank/patpass/PatpassOptions.java

```diff
@@ -0,0 +1,55 @@
+package cl.transbank.patpass;
+
+import cl.transbank.common.IntegrationType;
+import cl.transbank.common.Options;
+import lombok.*;
+
+@ToString  @AllArgsConstructor
+public class PatpassOptions extends Options implements Cloneable  {
+    @Getter  final String headerCommerceCodeName = "commercecode";
+    @Getter  final String headerApiKeyName = "Authorization";
+
+    public Options buildOptions(Options patpassOptions) {
+        PatpassOptions alt = clone();
+
+        // If the method receives an options object then rewrite each property, this is mandatory
+        if (null != patpassOptions) {
+            if (null != patpassOptions.getCommerceCode() && !patpassOptions.getCommerceCode().trim().isEmpty())
+                alt.setCommerceCode(patpassOptions.getCommerceCode());
+
+            if (null != patpassOptions.getApiKey() && !patpassOptions.getApiKey().trim().isEmpty())
+                alt.setApiKey(patpassOptions.getApiKey());
+
+            if (null != patpassOptions.getIntegrationType())
+                alt.setIntegrationType(patpassOptions.getIntegrationType());
+        }
+
+        return alt;
+    }
+
+    public PatpassOptions(String commerceCode, String apiKey,IntegrationType integrationType ){
+        super(commerceCode,apiKey,integrationType);
+    }
+
+    @Override
+    public PatpassOptions clone() {
+        try {
+            return (PatpassOptions) super.clone();
+        } catch (CloneNotSupportedException e) {
+            return null;
+        }
+    }
+
+    public boolean isEmpty() {
```

Comment:
This both method `isEmpty` could be implemented in the superclass `Options` too

Reply:
done

Reply:
done